src/app/modules/authorization/authorization.reducer.ts

```diff
@@ -7,35 +7,63 @@ export const authorizationFeatureKey = 'authorization';
 
 export interface State {
   authorizationToken?: string;
+  processing: {
+    signIn: boolean,
+    signUp: boolean,
+    signOut: boolean,
+  };
 }
 
 const authorizationTokenStorageKey = 'authorization';
 
 export const initialState: State = {
   authorizationToken: localStorage.getItem(authorizationTokenStorageKey) || undefined,
+  processing: {
+    signIn: false,
+    signUp: false,
+    signOut: false,
+  }
 };
 
 const authorizationReducer = createReducer(
   initialState,
-  on(
-    AuthorizationActions.signUpSuccess,
-    AuthorizationActions.signInSuccess,
+  on(AuthorizationActions.signInSuccess,
     (state, { authorizationToken }) => {
       localStorage.setItem(authorizationTokenStorageKey, authorizationToken);
-      return { ...state, authorizationToken };
+      return { ...state, authorizationToken, processing: { ...state.processing, signIn: false } };
+    }),
+  on(AuthorizationActions.signUpSuccess,
+    (state, { authorizationToken }) => {
+      localStorage.setItem(authorizationTokenStorageKey, authorizationToken);
+      return { ...state, authorizationToken, processing: { ...state.processing, signUp: false } };
+    }),
+  on(AuthorizationActions.signOutSuccess,
+    state => {
+      localStorage.removeItem(authorizationTokenStorageKey);
+      return { ...state, authorizationToken: undefined, processing: { ...state.processing, signOut: false } };
     }
```

Comment:
rozumiem, że te flagi processing to jest ze "trwa proces" i dlatego resetujesz flagi na kazda akcje?